### DISCUSSION TITLE
How to access link properties via a backlink?

### DISCUSSION THREAD

**Discussion by HamishWHC** [geldata/gel#4916] [Q&A] [answered]
Hi, I've been tinkering with EdgeDB today and thinking about using it for a new project, but have been encountering a weird issue with trying to access link properties via backlinks.

Here's my schema:
```
function now() -> cal::local_datetime
    using (cal::to_local_datetime(datetime_of_transaction(), "UTC"));

type Asset {
    required property serial_number -> str;

    multi link assignments -> AssignmentTarget {
        property began_at -> cal::local_datetime {
            default := now();
        };
        property ended_at -> cal::local_datetime;
        property current := exists @began_at and not exists @ended_at;
    };
    single link current_assignment := (
        select .assignments
        filter @current
        limit 1
    );
}

abstract type AssignmentTarget {
    multi link assignments := .<assignments[is Asset];
    multi link current_assignments := (
        select .<assignments[is Asset]
        filter @current
    );
}

type Customer extending AssignmentTarget {
    required property first_name -> str;
    required property last_name -> str;
    required property full_name := .first_name ++ " " ++ .last_name;
}
```

The following query works fine:
```
select Asset {
  serial_number,
  assignments: {
    [is Customer].first_name,
    @began_at,
    @current
  }
}
```

But this one doesn't:
```
select Customer {
  first_name,
  assignments: {
    serial_number,
    @began_at,
    @current
  },
}
```
It raises `InvalidReferenceError: link 'assignments' of object type 'default::Customer' has no property 'current'`. Not sure why this is happening considering I can filter by `@current`. Why is this error occurring (and where's it documented, I feel like I've seen some bit of the docs say this doesn't work but I can't find it again), and is there a way to achieve this without resorting to having an intermediate type (like you would in a regular SQL schema)?

Thanks for your time and help, and for building this awesome DB.

**Comment by raddevon** [comment, marked as answer]
I think you are running up against a known bug: https://github.com/edgedb/edgedb/issues/4825

**Reply by HamishWHC** [reply]
Ah, hadn't found that issue. Thanks for the help. Good to know it's a bug, I'll query manually until it's fixed.
